iModelCore/ECDb/ECDb/ECSql/ECSqlPreparer.cpp

```diff
@@ -1972,6 +1985,505 @@ ECSqlStatus ECSqlExpPreparer::PrepareTypeListExp(NativeSqlBuilder::List& nativeS
+    {
+    ECSqlStatus status;
+    for (size_t nPos = 0; nPos < exp.GetChildrenCount(); nPos++)
+        {
+        NativeSqlBuilder nativeSqlBuilder;
+        status = PrepareWindowDefinitionExp(nativeSqlBuilder, ctx, exp.GetChildren()[nPos]->GetAs<WindowDefinitionExp>());
+        if (!status.IsSuccess())
+            return status;
+
+        nativeSqlBuilderSnippets.push_back(nativeSqlBuilder);
+        }
+
+    return ECSqlStatus::Success;
+    }
+
+//-----------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+------
+//static`
+ECSqlStatus ECSqlExpPreparer::PrepareWindowDefinitionExp(NativeSqlBuilder& nativeSqlBuilder, ECSqlPrepareContext& ctx, WindowDefinitionExp const& exp)
+    {
+    nativeSqlBuilder.Append(exp.GetWindowName()).Append(" AS ");
+    ECSqlStatus status = PrepareWindowSpecification(nativeSqlBuilder, ctx, *exp.GetWindowSpecification());
+    if (!status.IsSuccess())
+        return status;
+
+    return ECSqlStatus::Success;
+    }
+
+//-----------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+------
+//static`
+ECSqlStatus ECSqlExpPreparer::PrepareWindowFunctionExp(NativeSqlBuilder::List& nativeSqlSnippets, ECSqlPrepareContext& ctx, WindowFunctionExp const& exp)
+    {
+    ECSqlStatus status;
+    NativeSqlBuilder nativeSqlBuilder;
+    NativeSqlBuilder::List nativeSqlFunctionSnippets;
+    status = PrepareFunctionCallExp(nativeSqlFunctionSnippets, ctx, exp.GetWindowFunctionCallExp()->GetAs<FunctionCallExp>());
+    if (!status.IsSuccess())
+        return status;
+    
+    for (auto snippet : nativeSqlFunctionSnippets)
+        nativeSqlBuilder.Append(snippet);
+
+    if (FilterClauseExp const * e = exp.GetFilterClauseExp())
+        {
+        status = PrepareFilterClauseExp(nativeSqlBuilder, ctx, *e);
+        if (!status.IsSuccess())
+            return status;
+        }
+
+    nativeSqlBuilder.Append(" OVER");
+    if (WindowSpecification const * e = exp.GetWindowSpecification())
+        {
+        status = PrepareWindowSpecification(nativeSqlBuilder, ctx, *e);
+        if (!status.IsSuccess())
+            return status;
+        
+        nativeSqlSnippets.push_back(nativeSqlBuilder);
+        return ECSqlStatus::Success;
+        }
+    else if (exp.GetWindowName().size() != 0)
+        {
+        nativeSqlBuilder.AppendSpace();
+        nativeSqlBuilder.Append(exp.GetWindowName());
+        nativeSqlSnippets.push_back(nativeSqlBuilder);
+        return ECSqlStatus::Success;
+        }
+    else
+        {
+        ctx.Issues().Report(IssueSeverity::Error, IssueCategory::BusinessProperties, IssueType::ECSQL, "Unsupported window function expression.");

```

Comment:
You'll have to create IssueIds for each of the logged messages since this has been added on the main branch meanwhile

Reply:
added